lib/GenomeAnnotationAPI/GenomeAnnotationUtil.py

```diff
@@ -0,0 +1,311 @@
+                "feature_notes": "",
+                "feature_inference": "",
+                "feature_quality_warnings": []
+            }
+            if 'functions' in x:
+                f['feature_function'] = "; ".join(x.get('functions', [])),
+
+            if "location" in x:
+                f["feature_locations"] = [{"contig_id": loc[0],
+                                           "start": loc[1],
+                                           "strand": loc[2],
+                                           "length": loc[3]} for loc in x['location']]
+            else:
+                f["feature_locations"] = []
+
+            if 'dna_sequence' in x:
+                f["feature_dna_sequence"] = x['dna_sequence']
+
+                if 'md5' in x:
+                    f["feature_md5"] = x['md5']
+                else:
+                    f["feature_md5"] = hashlib.md5(x["dna_sequence"].upper()).hexdigest()
+            else:
+                f["feature_dna_sequence"] = ""
+                f["feature_md5"] = ""
+
+            if 'dna_sequence_length' in x:
+                f["feature_dna_sequence_length"] = x['dna_sequence_length']
+            else:
+                f["feature_dna_sequence_length"] = -1
+
+            f["feature_aliases"] = {}
+            if 'aliases' in x:
+                for key in x['aliases']:
+                    if isinstance(key, list):
+                        key = ':'.join(key)
+                    f["feature_aliases"][key] = []
+
+            if "feature_quality_score" in x:
+                f["feature_quality_score"] = str(x['quality'])
+            else:
+                f["feature_quality_score"] = ""
+
+            return f
+
+        if feature_id_list is None:
+            for x in features:
+                out_features[x['id']] = fill_out_feature(x)
+        else:
+            if not isinstance(feature_id_list, list):
+                raise TypeError("A list of strings indicating Feature identifiers is required.")
+            try:
+                feature_refs = ["features/" + x for x in feature_id_list]
+                assert len(feature_refs) > 0
+            except TypeError:
+                raise TypeError("A list of strings indicating Feature identifiers is required.")
+            except AssertionError:
+                raise TypeError("A list of strings indicating Feature identifiers is required, received an empty list.")
+
+            for x in features:
+                if x['id'] in feature_id_list:
+                    out_features[x['id']] = fill_out_feature(x)
+
+        return out_features
+
+    def get_cds_by_gene(self, genome_ref, gene_id_list=None):
+        cds_by_gene = {}
+        genome = self.get_genome(genome_ref)
+        id_set = set(gene_id_list)
+        for gene in genome.get('features', []):
+            if id_set and gene['id'] not in id_set:
+                continue
+            if 'cdss' in gene:
+                cds_by_gene[gene['id']] = gene['cdss']
```

Comment:
I assume genome['features']['id'] are all different? I think it would be the case since it's called 'id' but I don't think it's enforced. I don't if we want to consider cases when features['id'] are the same but features['cdss'] are not.

Reply:
I suppose I could make cds_by_gene a set that is updated to collect all unique values but really we should not have duplicate ids